runtime/drivers/bigquery/olap.go

```diff
@@ -0,0 +1,536 @@
+package bigquery
+
+import (
+	"context"
+	sqldriver "database/sql/driver"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"math/big"
+	"reflect"
+	"time"
+
+	"cloud.google.com/go/bigquery"
+	"cloud.google.com/go/civil"
+	runtimev1 "github.com/rilldata/rill/proto/gen/rill/runtime/v1"
+	"github.com/rilldata/rill/runtime/drivers"
+	"github.com/rilldata/rill/runtime/pkg/observability"
+	"go.uber.org/zap"
+	"google.golang.org/api/iterator"
+)
+
+var _ drivers.OLAPStore = (*Connection)(nil)
+
+// Dialect implements drivers.OLAPStore.
+func (c *Connection) Dialect() drivers.Dialect {
+	return drivers.DialectBigQuery
+}
+
+// Exec implements drivers.OLAPStore.
+func (c *Connection) Exec(ctx context.Context, stmt *drivers.Statement) error {
+	if c.config.LogQueries {
+		c.logger.Info("bigquery query", zap.String("sql", c.Dialect().SanitizeQueryForLogging(stmt.Query)), zap.Any("args", stmt.Args), observability.ZapCtx(ctx))
+	}
```

Comment:
Can `Exec` be implemented as a wrapper on `Query` to avoid duplication of the query logging and other logic? Similar to DuckDB: https://github.com/rilldata/rill/blob/d7361a4245f681b0ff0a54b870aaf893de313700/runtime/drivers/duckdb/olap.go#L56-L66